deployability/modules/allocation/vagrant/utils.py

```diff
@@ -100,9 +111,10 @@ def get_port(cls, remote_host_parameters: dict, arch: str = None) -> int:
 
             raise ValueError(f"ppc64 server has no available SSH ports.")
         else:
-            for i in range(20, 40):
-                port = f"432{i}"
+            for i in range(20):
+                port = f"432{random.randint(20, 40)}"
```

Comment:
We need to improve this, it was checking ports in the range from 20 to 40, with this approach, it will test random ports between 20-40, which could lead to checking the same port multiple times when it shouldn't

As discussed,  we could use an exclusion list to avoid rechecking ports, this will test all ports between 20 and 40 but will allow us to force differents ports when multiples instances are being launched at the same time

Reply:
Done: 94bb6b845

Test:

```console
cbordon@cbordon-MS-7C88:~/Downloads/wazuh-qa$ python3 deployability/modules/allocation/main.py --provider vagrant --size small --instance-name jose-carreras-test --composite-name macos-sonoma-14.4.1-amd64
[2024-05-10 11:37:24] [INFO] ALLOCATOR: Creating instance at /tmp/wazuh-qa
[2024-05-10 11:37:36] [INFO] ALLOCATOR: Using the macStadium Intel server to deploy.
[2024-05-10 11:37:36] [DEBUG] ALLOCATOR: Checking if instance directory exists on remote host
[2024-05-10 11:37:39] [DEBUG] ALLOCATOR: Creating instance directory on remote host
[2024-05-10 11:37:41] [DEBUG] ALLOCATOR: No config provided. Generating from payload
[2024-05-10 11:37:41] [DEBUG] ALLOCATOR: Generating new key pair
[2024-05-10 11:37:48] [DEBUG] ALLOCATOR: Vagrantfile created. Creating instance.
[2024-05-10 11:37:58] [INFO] ALLOCATOR: Instance jose-carreras-test-1092 created.
[2024-05-10 11:38:44] [INFO] ALLOCATOR: Instance jose-carreras-test-1092 started.
[2024-05-10 11:38:56] [INFO] ALLOCATOR: Inventory file generated at /tmp/wazuh-qa/jose-carreras-test-1092/inventory.yml
[2024-05-10 11:38:58] [INFO] ALLOCATOR: SSH connection successful.
[2024-05-10 11:39:10] [INFO] ALLOCATOR: Track file generated at /tmp/wazuh-qa/jose-carreras-test-1092/track.yml
cbordon@cbordon-MS-7C88:~/Downloads/wazuh-qa$ cat /tmp/wazuh-qa/jose-carreras-test-1092/inventory.yml
ansible_connection: ssh
ansible_host: 207.254.50.205
ansible_password: vagrant
ansible_port: 43226
ansible_ssh_common_args: -o StrictHostKeyChecking=no
ansible_user: vagrant
cbordon@cbordon-MS-7C88:~/Downloads/wazuh-qa$ python3 deployability/modules/allocation/main.py --provider vagrant --size small --instance-name jose-carreras-test --composite-name macos-sonoma-14.4.1-amd64
[2024-05-10 11:40:43] [INFO] ALLOCATOR: Creating instance at /tmp/wazuh-qa
[2024-05-10 11:40:54] [INFO] ALLOCATOR: Using the macStadium Intel server to deploy.
[2024-05-10 11:40:54] [DEBUG] ALLOCATOR: Checking if instance directory exists on remote host
[2024-05-10 11:40:57] [DEBUG] ALLOCATOR: Creating instance directory on remote host
[2024-05-10 11:40:59] [DEBUG] ALLOCATOR: No config provided. Generating from payload
[2024-05-10 11:40:59] [DEBUG] ALLOCATOR: Generating new key pair
[2024-05-10 11:41:05] [DEBUG] ALLOCATOR: Vagrantfile created. Creating instance.
[2024-05-10 11:41:15] [INFO] ALLOCATOR: Instance jose-carreras-test-8666 created.
[2024-05-10 11:42:01] [INFO] ALLOCATOR: Instance jose-carreras-test-8666 started.
[2024-05-10 11:42:13] [INFO] ALLOCATOR: Inventory file generated at /tmp/wazuh-qa/jose-carreras-test-8666/inventory.yml
[2024-05-10 11:42:14] [INFO] ALLOCATOR: SSH connection successful.
[2024-05-10 11:42:25] [INFO] ALLOCATOR: Track file generated at /tmp/wazuh-qa/jose-carreras-test-8666/track.yml
cbordon@cbordon-MS-7C88:~/Downloads/wazuh-qa$ cat /tmp/wazuh-qa/jose-carreras-test-8666/inventory.yml
ansible_connection: ssh
ansible_host: 207.254.50.205
ansible_password: vagrant
ansible_port: 43221
ansible_ssh_common_args: -o StrictHostKeyChecking=no
ansible_user: vagrant
```

Reply:
Improved validation to avoid an infinite loop: 3430a10b2